drivers/power_domain/power_domain_psl.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_power_domain_psl
+
+#include <kernel.h>
+#include <drivers/gpio.h>
+#include <drivers/pinctrl.h>
+#include <pm/device.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(power_domain_psl, LOG_LEVEL_INF);
+
+struct npcx_psl_config {
+	struct gpio_dt_spec enable;
+	/* PSL_IN pinmux configuration */
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct npcx_psl_data {
+	int event;
+};
+
+const char *actions[] = {
+	[PM_DEVICE_ACTION_TURN_OFF] = "TURN OFF"
+};
+
+static int npcx_psl_pm_action(const struct device *dev,
+			     enum pm_device_action action)
+{
+	const struct npcx_psl_config *config = dev->config;
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_TURN_OFF:
+		/* Configure detection settings of PSL_IN pads first */
+		ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+		if (ret < 0) {
+			LOG_ERR("PSL_IN pinctrl setup failed (%d)", ret);
+			return ret;
+		}
+
+		/* Turn the Core Domain power supply (VCC1) off */
+		gpio_pin_set_dt(&config->enable, 1);
+		break;
+	default:
+		ret = -ENOTSUP;
+	}
+
+	return ret;
+}
+
+static int npcx_psl_init(const struct device *dev)
+{
+	const struct npcx_psl_config *config = dev->config;
+
+	if (!device_is_ready(config->enable.port)) {
+		LOG_ERR("GPIO port %s is not ready", config->enable.port->name);
+		return -ENODEV;
+	}
+
```

Comment:
Should there be a gpio_pin_configure_dt() somewhere here?

Reply:
No need for ```gpio_pin_configure_dt()```. GPIO85 is not a standard GPIO and PSL_OUT is switched to ```inactive``` state only by setting bit 7 in PDOUT reg of gpio8 module. The following is a brief introduction in our datasheet.

_GPIO85 is used as the PSL_OUT signal. This pin does not have output enable control, open-drain capability, weak pull-up, weak pull-down or “Drive Enable by VDD Present”. Therefore, the PxOTYPE, PxPULL, PxPUD and PxENVDD bits must be set to 0; the value of PxDIR bit is ignored. If this pin is not used for PSL functionality, it must be left unconnected._

Reply:
Oh, fancy... Would it be worth adding that paragraph here as a comment?

Reply:
Done.